### ISSUE TITLE
Upgrade fails with missing class

### ISSUE THREAD

**Issue by shadowhand** [CaptainHookPhp/captainhook#65]
After upgrading from CaptainHook `4.5.4` to `4.5.8`, the following error appears on existing checkouts:

```
Fatal error: Uncaught Error: Class 'SebastianFeldmann\Camino\Path\Directory' not found in vendor/captainhook/captainhook/src/Hook/Template/Builder.php:60
```

The `sebastianfeldmann/camino` package is being installed as part of the update, and the autoloader has not yet updated, so the post-update hooks fail. This issue is breaking our builds and causing some developer confusion.

**Comment by sebastianfeldmann** [comment]
Hi

thanks for reporting this.
I'm not sure how to solve this.
Currently I'm doing something like this.

```php
if (!$this->isCaptainHookInstalled()) {
    // reload the autoloader to make sure CaptainHook is available
    $vendorDir = $event->getComposer()->getConfig()->get('vendor-dir');
    require $vendorDir . '/autoload.php';
}
```
So if the Cap`n is not found I reload the autoloader.
It feels wrong to do this every time just to be sure.
If other plugins would do this as well it will most likely crash as well.

I have to check if there is some composer functionality that I'm missing so far

**Comment by sebastianfeldmann** [comment]
The GrumPHP guys solved this problem in an interesting way, they are calling their own binary in a sub process. This way they don't have to deal with any inconsistent autoloading state.

I will have a closer look at their solution any most likely choose a similar approach to solve this issue.

**Comment by sebastianfeldmann** [comment]
Fixed in version 5.0.0 currently only available as RC2 :(
Hopefully coming soon
